src/Microsoft.AspNetCore.Localization/QueryStringRequestCultureProvider.cs

```diff
@@ -71,9 +76,34 @@ public override Task<ProviderCultureResult> DetermineProviderCultureResult(HttpC
                 queryCulture = queryUICulture;
             }
 
+            if (!IsValidCulture(queryCulture))
+            {
+                _logger = _logger ?? httpContext.RequestServices.GetService<ILogger<QueryStringRequestCultureProvider>>();
```

Comment:
Ah, I though from your comment in the PR #385 that you gonna pass it in the constructor, seems the idea changed little bit